pkg/agent/config.go

```diff
@@ -258,151 +268,430 @@ func InitAgentCmdFlags(c *cobra.Command, cfg *AgentCmdFlags) {
+	OrganizationID string
+	EndpointPath   string // Deprecated.
+
+	// VenafiCloudKeypair mode only.
+	UploadPath         string
+	ClusterDescription string
+
+	// VenafiCloudVenafiConnection mode only.
+	VenConnName string
+	VenConnNS   string
+	InstallNS   string
+
+	// Only used for testing purposes.
+	OutputPath string
+	InputPath  string
+}
 
-	// In VenafiConnection mode, we don't need the server field. For the other
-	// modes, we do need to validate the server field.
-	var baseURL string
-	if flags.VenConnName != "" {
-		if cfg.Server != "" {
-			log.Printf("ignoring the server field specified in the config file. In Venafi Connection mode, this field is not needed. Use the VenafiConnection's spec.vcp.url field instead.")
+// ValidateAndCombineConfig combines and validates the input configuration with
+// the flags passed to the agent and returns the final configuration as well as
+// the Venafi client to be used to upload data. Does not do any network call.
+// The logger can be changed for testing purposes. You do not need to call
+// ValidateDataGatherers as ValidateAndCombineConfig already does that.
+//
+// The error returned may be a multierror.Error. Use multierror.Prefix(err,
+// "context:") rather than fmt.Errorf("context: %w", err) when wrapping the
+// error.
+func ValidateAndCombineConfig(log *log.Logger, cfg Config, flags AgentCmdFlags) (CombinedConfig, client.Client, error) {
+	res := CombinedConfig{}
+	var errs error
+
+	{
+		var mode AuthMode
+		switch {
+		case flags.VenafiCloudMode && flags.CredentialsPath != "":
+			mode = VenafiCloudKeypair
+			log.Printf("Using the %s auth mode since --venafi-cloud and --credentials-path were specified.", mode)
+		case flags.ClientID != "" && flags.PrivateKeyPath != "":
+			mode = VenafiCloudKeypair
+			log.Printf("Using the %s auth mode since --client-id and --private-key-path were specified.", mode)
+		case flags.ClientID != "":
+			return CombinedConfig{}, nil, fmt.Errorf("if --client-id is specified, --private-key-path must also be specified")
+		case flags.PrivateKeyPath != "":
+			return CombinedConfig{}, nil, fmt.Errorf("--private-key-path is specified, --client-id must also be specified")
+		case flags.VenConnName != "":
+			mode = VenafiCloudVenafiConnection
+			log.Printf("Using the %s auth mode since --venafi-connection was specified.", mode)
+		case flags.APIToken != "":
+			mode = JetstackSecureAPIToken
+			log.Printf("Using the %s auth mode since --api-token was specified.", mode)
+		case !flags.VenafiCloudMode && flags.CredentialsPath != "":
+			mode = JetstackSecureOAuth
+			log.Printf("Using the %s auth mode since --credentials-file was specified without --venafi-cloud.", mode)
+		default:
+			return CombinedConfig{}, nil, fmt.Errorf("no auth mode specified. You can use one of four auth modes:\n" +
+				" - Use --venafi-cloud with --credentials-file or --client-id with --private-key-path to use the " + string(VenafiCloudKeypair) + " mode.\n" +
```

Comment:
```suggestion
				" - Use (--venafi-cloud with --credentials-file) or (--client-id with --private-key-path) to use the " + string(VenafiCloudKeypair) + " mode.\n" +
```
Somehow it took me a long time to understand this line without the parentheses.

Reply:
Good point, adding parentheses will make it much easier to read.